// react-native-iap/lib/commonjs/modules/iosSk2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.sync = exports.subscriptionStatus = exports.showManageSubscriptions = exports.latestTransaction = exports.isEligibleForIntroOffer = exports.finishTransaction = exports.currentEntitlement = exports.beginRefundRequest = void 0;
var _reactNative = require("react-native");
const {
  RNIapIosSk2
} = _reactNative.NativeModules;
/**
 * Sync state with Appstore (iOS only)
 * https://developer.apple.com/documentation/storekit/appstore/3791906-sync
 */
const sync = () => RNIapIosSk2.sync();

/**
 *
 */
exports.sync = sync;
const isEligibleForIntroOffer = groupID => RNIapIosSk2.isEligibleForIntroOffer(groupID);

/**
 *
 */
exports.isEligibleForIntroOffer = isEligibleForIntroOffer;
const subscriptionStatus = sku => RNIapIosSk2.subscriptionStatus(sku);

/**
 *
 */
exports.subscriptionStatus = subscriptionStatus;
const currentEntitlement = sku => RNIapIosSk2.currentEntitlement(sku);

/**
 *
 */
exports.currentEntitlement = currentEntitlement;
const latestTransaction = sku => RNIapIosSk2.latestTransaction(sku);

/**
 *
 */
exports.latestTransaction = latestTransaction;
const beginRefundRequest = sku => RNIapIosSk2.beginRefundRequest(sku);

/**
 *
 */
exports.beginRefundRequest = beginRefundRequest;
const showManageSubscriptions = () => RNIapIosSk2.showManageSubscriptions();

/**
 *
 */
exports.showManageSubscriptions = showManageSubscriptions;
const finishTransaction = transactionIdentifier => RNIapIosSk2.finishTransaction(transactionIdentifier);
exports.finishTransaction = finishTransaction;
